### PR TITLE
NMS-13075: WmiDetector documentation

### PR DESCRIPTION
**Pull request by mfuhrmann**
### All Contributors

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue which points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-13075

### CODE REVIEW

**Comment by mfuhrmann** [comment]
I've collected information from the Wiki to create this PR. Unfortunately, I don't have a working example to add here.

File: docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
+= WMI Service Detector
+
+The WMI Detector can be used to detect the availability of WmiConfiguration on a device when scanning it using Provisiond. By default, it expects to make a WMI query to the `Win32_ComputerSystem` class, checking for `OK` from the "Status" object.
```
Comment (indigo423) [inline docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc:3]: one line one sentence

**Review by indigo423 — APPROVED** [review]
just the copy & paste formatting fix. 👍

File: docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc (hunk 1 above)
Comment (mfuhrmann) [reply docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc]: fixed

Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc]:
```suggestion
Use the WMI Detector to detect the availability of WmiConfiguration on a device when scanning it using Provisiond. By default, it expects to make a WMI query to the `Win32_ComputerSystem` class, checking for `OK` from the "Status" object.
```

File: docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+== Configuration and Use
+
+.Parameters for the Win32 Service Detector
+[options="header, autowidth"]
+|===
+| Parameter       | Description                                                                              | Required | Default value
+| `username`   | The username to use when authenticating against the WMI host.  | Required | `Administrator`
+| `password`   | The password to use when authenticating against the WMI host.  | Required |
+| `domain`   | The Windows domain to use when authenticating against the WMI host.  | Required | `WORKGROUP`
+| `wmiClass`   | This is the class of instances within WMI. An example of this would be the `Win32_ComputerSystem` class. This class contains objects representing properties and methods available.  | Required | `Win32_ComputerSystem`
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc:22]:
```suggestion
| `wmiClass`   | The class of instances within WMI. An example would be the `Win32_ComputerSystem` class. This class contains objects representing properties and methods available.  | Required | `Win32_ComputerSystem`
```

File: docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+| `wmiObject`   | A WMI Object is technically a member of a WMI property set. Using the previous example of `Win32_ComputerSystem` this class contains a variety of properties that we can look at and poll. The default WMI service uses the `Status` property to determine if the system is running and if WMI is available. | Required | `Status`
+| `wmiWqlStr`   | Microsoft WMI also implements an `ExecQuery` method which is more common in Windows scripting but less common in OpenNMS polling and monitoring. WQL, which stands for "SQL For WMI" according to Microsoft, is a simple SQL-like syntax for querying WMI classes, properties and instances. Here's an example of a WQL query that you could use to discover whether a Windows service is running: `Select State From Win32_Service Where Name='Server'` | Required | `NOTSET`
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc:24]:
```suggestion
| `wmiWqlStr`   | Microsoft WMI also implements an `ExecQuery` method which is more common in Windows scripting but less common in OpenNMS polling and monitoring. WQL, which stands for "SQL For WMI" according to Microsoft, is a simple SQL-like syntax for querying WMI classes, properties, and instances. Here's an example of a WQL query that you could use to discover whether a Windows service is running: `Select State From Win32_Service Where Name='Server'` | Required | `NOTSET`
```

File: docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+| `compVal`   | The value to use when performing a comparison against the wmiObject. | Required | `OK`
+| `compOp`   a| The compare operation is how you tell the OpenNMS WMI poller plugins how to verify the nature (up, critical) of a specific property within WMI. In the event of unequal comparisons the system will always have the WMI value (as retrieved from the target) on the left. Available compare operations are:
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc:26]:
```suggestion
| `compOp`   a| The compare operation is how you tell the OpenNMS WMI poller plugins how to verify the nature (up, critical) of a specific property within WMI. In the event of unequal comparisons, the system will always have the WMI value (as retrieved from the target) on the left. Available compare operations are:
```

File: docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+* `EQ` (equals)
+* `NEQ` (not equals)
+* `GT` (greater than)
+* `LT` (less than)
+* `NOOP` (no operation)
+
+The NOOP operation is a special case in which the Manager will always return a result code of "OK." This is typical used in scenarios where the existence of a class or property is more interesting than the contents of the property. We use this in the collector to check and ensure that WMI is available before attempting to collect data. | Required | `EQ`
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc:34]:
```suggestion
The NOOP operation is a special case in which the Manager will always return a result code of "OK". This is typically used in scenarios where the existence of a class or property is more interesting than the contents of the property. We use this in the collector to check and ensure that WMI is available before attempting to collect data. | Required | `EQ`
```

File: docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+| `matchType`   a| The match type configuration tells the OpenNMS WMI system how you want to cope with multiple instance results. Available match types are:
+
+* `all` all instances must comply to the compare operation
+* `none` no instances should comply with the compare operation
+* `some` - only some (1 or more) must comply with the compare operation
+* `one` only one instance can comply with the compare operation. | Required | `all`
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc:40]:
```suggestion
* `one` only one instance can comply with the compare operation | Required | `all`
```

File: docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+.Parameters for the Win32 Service Detector
+[options="header, autowidth"]
+|===
+| Parameter       | Description                                                                              | Required | Default value
+| `username`   | The username to use when authenticating against the WMI host.  | Required | `Administrator`
+| `password`   | The password to use when authenticating against the WMI host.  | Required |
+| `domain`   | The Windows domain to use when authenticating against the WMI host.  | Required | `WORKGROUP`
+| `wmiClass`   | This is the class of instances within WMI. An example of this would be the `Win32_ComputerSystem` class. This class contains objects representing properties and methods available.  | Required | `Win32_ComputerSystem`
+| `wmiObject`   | A WMI Object is technically a member of a WMI property set. Using the previous example of `Win32_ComputerSystem` this class contains a variety of properties that we can look at and poll. The default WMI service uses the `Status` property to determine if the system is running and if WMI is available. | Required | `Status`
```
Comment (Bonrob2) [inline docs/modules/operation/pages/provisioning/detectors/WmiDetector.adoc:23]:
```suggestion
| `wmiObject`   | A WMI object is technically a member of a WMI property set. Using the previous example of `Win32_ComputerSystem` this class contains a variety of properties that we can look at and poll. The default WMI service uses the `Status` property to determine if the system is running and if WMI is available. | Required | `Status`
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
Just a few changes. Thanks for all the work.

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
Looks good. Just a few changes.

**Pull request merged**